source/nutrition/nutrition_facts.h

```diff
@@ -91,6 +91,16 @@ namespace nutrition {
         // Convenience operator /=
         nutrition_facts& operator/=(double scalar);
 
+      private:
+        /// Invariants
+        // No zero and negative for operator/=
+        void invariant_no_zero_and_negative(const nutrition_facts& other_to_check);
+        bool invariant_no_zero_and_negative(double scalar);
+        // No negative value for operator*= (allow zeros)
+        bool invariant_no_negative_only(double value);
+        void invariant_no_negative_only(const nutrition_facts& other_to_check);
+        // No negative result value for operator-=
+        void invariant_no_negative_result_only(const nutrition_facts& result);
```

Comment:
Funções invariantes são 1) privadas, 2) const, 3) sem parâmetros e 4) sempre retornam bool.

Se quiser testar "other", é só chamar other->invariants(), mas como disse, chamamos essa função apenas para o this.

Também não precisamos criar várias funções dessas. Apenas 1 para ver se não quebramos nada com o objeto e chamamos sempre essa no final das chamadas de funções mutáveis.